torch_geometric/explain/algorithm/subgraphX_explainer.py

```diff
@@ -0,0 +1,646 @@
+        info_dict = {
+            "data": self.data.to("cpu"),
+            "coalition": self.coalition,
+            "ori_graph": self.ori_graph,
+            "W": self.W,
+            "N": self.N,
+            "P": self.P,
+        }
+        return info_dict
+
+    def load_info(self, info_dict):
+        self.W = info_dict["W"]
+        self.N = info_dict["N"]
+        self.P = info_dict["P"]
+        self.coalition = info_dict["coalition"]
+        self.ori_graph = info_dict["ori_graph"]
+        self.data = info_dict["data"].to(self.device)
+        self.children = []
+        return self
+
+
+class MCTS(object):
+    r"""
+    Monte Carlo Tree Search Method.
+
+    Args:
+        X (:obj:`torch.Tensor`): Input node features
+        edge_index (:obj:`torch.Tensor`): The edge indices.
+        num_hops (:obj:`int`): Number of hops :math:`k`.
+        n_rollout (:obj:`int`): Number of sequences to build the
+            monte carlo tree.
+        min_atoms (:obj:`int`): Number of atoms for the subgraph
+            in the monte carlo tree leaf node.
+        c_puct (:obj:`float`): Hyper-parameter to encourage
+            exploration while searching.
+        expand_atoms (:obj:`int`): Number of children to expand.
+        high2low (:obj:`bool`): Whether to expand children tree node
+            from high degree nodes to low degree nodes.
+        node_idx (:obj:`int`): The target node index to extract
+            the neighborhood.
+        score_func (:obj:`Callable`): The reward function for
+            tree node, such as mc_shapely and mc_l_shapely.
+    """
+    def __init__(
+        self,
+        X: torch.Tensor,
+        edge_index: torch.Tensor,
+        num_hops: int,
+        n_rollout: int = 10,
+        min_atoms: int = 3,
+        c_puct: float = 10.0,
+        expand_atoms: int = 14,
+        high2low: bool = False,
+        node_idx: int = None,
+        score_func: Callable = None,
+        device="cpu",
+    ):
+
+        self.X = X
+        self.edge_index = edge_index
+        self.device = device
+        self.num_hops = num_hops
+        self.data = Data(x=self.X, edge_index=self.edge_index)
+        graph_data = Data(x=self.X,
+                          edge_index=remove_self_loops(self.edge_index)[0])
+        self.graph = to_networkx(graph_data, to_undirected=True)
+        self.data = Batch.from_data_list([self.data])
+        self.num_nodes = self.graph.number_of_nodes()
+        self.score_func = score_func
+        self.n_rollout = n_rollout
+        self.min_atoms = min_atoms
+        self.c_puct = c_puct
+        self.expand_atoms = expand_atoms
+        self.high2low = high2low
+        self.new_node_idx = None
+
+        # extract the sub-graph and change the node indices.
+        if node_idx is not None:
+            if isinstance(node_idx, torch.Tensor):
+                node_idx = node_idx.item()
+
+            self.ori_node_idx = node_idx
+            self.ori_graph = copy.copy(self.graph)
```

Comment:
Not sure what others think but my preference would be to have longer, more descriptive variable names like `original_node_idx` or `original_graph`.